pkg/utils/machine/machine.go

```diff
@@ -15,18 +15,118 @@ limitations under the License.
 package machine
 
 import (
+	"context"
+	"errors"
+	"fmt"
+
 	"github.com/samber/lo"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
 	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
 	"github.com/aws/karpenter-core/pkg/operator/scheme"
 	"github.com/aws/karpenter-core/pkg/scheduling"
 )
 
+func EventHandler(ctx context.Context, c client.Client) handler.EventHandler {
```

Comment:
Can you add comment on here to say what it's used for?
```suggestion
// EventHandler creates a mapping from a machine to a reconcile request for the linked node.
func EventHandler(ctx context.Context, c client.Client) handler.EventHandler {
```